### DISCUSSION TITLE
Winui版本问题

### DISCUSSION THREAD

**Discussion by 2ji3150** [Blinue/Magpie#799] [Dev Channel]
好奇问一下开发者。
在想能不能贡献，看了一下源代码似乎是winui2 （c++ / WinRT)
未來有改成winui3的打算吗？

_由于我的专长是c#的部分，不知道未来有没有考虑做API端口？
这样的话我可以写个命令行界面或是一个基于c#的winui3第三方界面做贡献。
有什么特殊理由使用c++的winrt的话我再继续学习一下。

**Comment by Blinue** [comment]
主要原因是 C# 和 WindowsAppSDK 会引入依赖项，增加部署难度。C# 需要部署 .NET 运行时，WindowsAppSDK 也需要部署 Windows App SDK runtime。如果让用户自己安装会增加使用门槛，Magpie 曾经依赖 .NET 运行时，事实证明很多用户不会安装运行时；如果把运行时一起打包会大大增加包的体积，WindowsAppSDK 应用的体积往往有 100 MB 以上。

我个人认为，WinAppSDK 已经失败了，很多迹象可以印证：
1. 微软的很多第一方软件依然是 UWP，或者转向了 XAML Islands 甚至 Web
2. 开发进度缓慢，社区无人管理，新旧 bug 堆积无人问津
3. 开发体验糟糕，缺失很多基础功能，至今依然没有 XAML 设计器
4. 用户体验也很糟糕，因为性能太差，包的体积也很大

基于这些原因，我暂时不考虑迁移到 WinUI 3。

> 不知道未来有没有考虑做API端口？

[Magpie.Core](https://github.com/Blinue/Magpie/tree/dev/src/Magpie.Core) 包含了核心的缩放功能，它本身是一个相对独立的组件，你可以把它编译成 dll，然后通过 P/Invoke 调用。

**Reply by 2ji3150** [reply]
感谢回复，我再试试。

**Reply by Shiiho11** [reply]
新手开始学客户端，发现确实如Blinue所说，winui3问题太多了，几乎什么都没有的模板示例程序，发布框架依赖未打包app，一堆DLL，体积30+MB。.net运行时和WindowsAppSDK也阻碍的了即开即用的体验。新手教程几乎没有，开发体验也很差。除了UI炫酷，其他真的不行。

**Reply by 2ji3150** [reply]
@Shiiho11  但问题是也找不到第二个好看&modern的windows GUI框架了。
